src/wallet/wallet.cpp

```diff
@@ -3950,8 +3952,13 @@ ScriptPubKeyMan* CWallet::AddWalletDescriptor(WalletDescriptor& desc, const Flat
         return nullptr;
     }
 
+    std::string error;
     auto spk_man = GetDescriptorScriptPubKeyMan(desc);
     if (spk_man) {
+        if (!spk_man->CanUpdateToWalletDescriptor(desc, error)) {
+            WalletLogPrintf("Invalid param: %s\n", error);
+            throw JSONRPCError(RPC_INVALID_PARAMETER, error);
```

Comment:
This introduces a cyclic dependency. Please do not add RPC dependencies to the wallet. These are two separate components, and the wallet operates at a lower level.